libraries/AP_NavEKF3/AP_NavEKF3_OptFlowFusion.cpp

```diff
@@ -289,8 +289,12 @@ void NavEKF3_core::FuseOptFlow(const of_elements &ofDataDelayed, bool really_fus
     ftype vd = stateStruct.velocity.z;
     ftype pd = stateStruct.position.z;
 
-    // constrain height above ground to be above range measured on ground
+    // constrain height above ground to be above range measured on ground, or user specified height (if given, for rover)
+#if APM_BUILD_TYPE(APM_BUILD_Rover)
+    ftype heightAboveGndEst = ofDataDelayed.heightOverride > 0 ? ofDataDelayed.heightOverride : MAX((terrainState - pd), rngOnGnd);
```

Comment:
As discussed on the dev call:

1. we shouldn't be correcting the range finder for the tilt of the vehicle
2. temporarily add some debug to be sure that the "range" variable below is being set to the FLOW_HEIGHT (or FLOW_HGR_OVR) parameter value.
3. we should understand why the EKF is not failing to fuse optical flow because it doesn't have a rangefinder.  We would have expected some checks would have needed to be changed.

Reply:
1 - fixed
2 - confirmed that the correct value is being sent through